core/src/scene/styleParam.h

```diff
@@ -60,6 +60,28 @@ constexpr size_t StyleParamKeySize = static_cast<size_t>(StyleParamKey::NUM_ELEM
 
 enum class Unit { pixel, milliseconds, meter, seconds };
 
+static std::string unitString(Unit unit) {
+    switch(unit) {
+        case Unit::pixel: return "pixel";
+        case Unit::milliseconds: return "milliseconds";
+        case Unit::meter: return "meter";
+        case Unit::seconds: return "seconds";
+        default: return "undefined";
+    }
+}
+
+template <typename T, int size>
+struct UnitVec {
+    T value = T(0.0);
+    Unit units[size];
```

Comment:
It actually is! At least in C++14. The values are all definitely known at compile-time, I think arithmetic might not have been allowed until C++14.